sdk/search/search-documents/src/searchIndexClient.ts

```diff
@@ -550,6 +603,114 @@ export class SearchIndexClient {
     }
   }
 
+  /**
+   * Creates a new search alias or updates an alias if it already exists.
+   * @param alias - The definition of the alias to create or update.
+   * @param options - The options parameters.
+   */
+  public async createOrUpdateAlias(
+    alias: SearchIndexAlias,
+    options: CreateOrUpdateAliasOptions = {}
+  ): Promise<SearchIndexAlias> {
+    const { span, updatedOptions } = createSpan("SearchIndexerClient-createOrUpdateAlias", options);
+    try {
+      const etag = options.onlyIfUnchanged ? alias.etag : undefined;
+
+      const result = await this.client.aliases.createOrUpdate(alias.name, alias, {
+        ...updatedOptions,
+        ifMatch: etag,
```

Comment:
Looking at the guidelines, my understanding is that the expectation is to set `if-none-match` when `onlyIfUnchanged` is true or `ifMatch` when `onlyIfChanged`

https://azure.github.io/azure-sdk/typescript_design.html#ts-conditional-requests